pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
Sorry, its still not entirely clear to me how to pass the actual indexer (level positioners). Are level positioners always designated as ints?

Going back to the example where the index names are also ints:

```
In [58]: s
Out[58]: 
1    0
one  a    1.0
     b    2.0
two  a    3.0
     b    4.0
dtype: float64
```

If I want to unstack the `a,b,a,b` level in the 1st position, `unstack(level=1)` will unstack the level in the 0th position instead of the 1st.